whisper/model.py

```diff
@@ -135,22 +136,33 @@ def forward(
         mask: Optional[Tensor] = None,
         kv_cache: Optional[dict] = None,
     ):
-        x = x + self.attn(self.attn_ln(x), mask=mask, kv_cache=kv_cache)
+        residual = x
+        x = self.attn(self.attn_ln(x), mask=mask, kv_cache=kv_cache)
+        x = nn.functional.dropout(x, p=self.dropout, training=self.training)
+        x = x + residual
         if self.cross_attn:
-            x = x + self.cross_attn(self.cross_attn_ln(x), xa, kv_cache=kv_cache)
+            residual = x
+            x = self.cross_attn(self.cross_attn_ln(x), xa, kv_cache=kv_cache)
+            x = nn.functional.dropout(x, p=self.dropout, training=self.training)
+            x = x + residual
+        # the MLP needs to be changed from: fc1 -> act_fn -> fc2 to: fc1 -> act_fn -> dropout -> fc2 -> dropout
+        # since it's written as an nn.Sequential, adding the dropout would change the state_dict
+        # instead, we add dropout by overriding our pre-trained model in our training script (much faster!)
         x = x + self.mlp(self.mlp_ln(x))
         return x
 
 
 class AudioEncoder(nn.Module):
-    def __init__(self, n_mels: int, n_ctx: int, n_state: int, n_head: int, n_layer: int):
+    def __init__(self, n_mels: int, n_ctx: int, n_state: int, n_head: int, n_layer: int, dropout: float):
         super().__init__()
+        self.dropout = dropout
+
         self.conv1 = Conv1d(n_mels, n_state, kernel_size=3, padding=1)
         self.conv2 = Conv1d(n_state, n_state, kernel_size=3, stride=2, padding=1)
         self.register_buffer("positional_embedding", sinusoids(n_ctx, n_state))
 
         self.blocks: Iterable[ResidualAttentionBlock] = nn.ModuleList(
-            [ResidualAttentionBlock(n_state, n_head) for _ in range(n_layer)]
+            [ResidualAttentionBlock(n_state, n_head, dropout=self.dropout) for _ in range(n_layer)]
```

Comment:
I wouldn't add dropout here because we freeze the encoder during training -> let's maybe disable it here?